pkg/server/server.go

```diff
@@ -298,23 +298,30 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 		opt(s)
 	}
 
-	s.checkOptions = []graph.LocalCheckerOption{
+	singleflightCheckResolver := graph.NewSingleflightCheckResolver()
```

Comment:
This is where the explicit delegate composition takes place with explicit `.SetDelegate` calls. It makes it abundantly clear which resolver is delegate to the other.